lib/channel.ml

```diff
@@ -90,23 +90,34 @@ let output_data t data =
     in
     Cstruct.fold (fun frags frag ->
         Ssh.Msg_channel_data (t.them.id, frag) :: frags)
-      i [] |> List.rev
+      i acc |> List.rev
   in
   let tosend = cs_join t.tosend data in
   let len = min (Cstruct.length tosend) (Int32.to_int t.them.win) in
-  let data, tosend = Cstruct.split tosend len in
+  let data, tosend =
+    if flush then
+      tosend, Cstruct.create 0
```

Comment:
Very minor nit: let's reuse the empty cstruct
```suggestion
      tosend, Cstruct.empty
```
I think `Cstruct.create 0` might do an (expensive-ish) allocation.